pkg/security/common/containerutils/utils.go

```diff
@@ -8,18 +8,41 @@ package containerutils
 
 import (
 	"regexp"
+	"strings"
 )
 
 // ContainerIDPatternStr defines the regexp used to match container IDs
-// ([0-9a-fA-F]{64}) is standard container id used pretty much everywhere
-// ([0-9a-fA-F]{32}-[0-9]{10}) is container id used by AWS ECS
-// ([0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){4}) is container id used by Garden
+// ([0-9a-fA-F]{64}) is standard container id used pretty much everywhere, lenght: 64
+// ([0-9a-fA-F]{32}-[0-9]{10}) is container id used by AWS ECS, lenght: 43
+// ([0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){4}) is container id used by Garden, lenght: 28
 var ContainerIDPatternStr = "([0-9a-fA-F]{64})|([0-9a-fA-F]{32}-[0-9]{10})|([0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){4})"
-
-// containerIDPattern is the pattern of a container ID
 var containerIDPattern = regexp.MustCompile(ContainerIDPatternStr)
 
+var containerIDCoreChars = "0123456789abcdefABCDEF"
+
 // FindContainerID extracts the first sub string that matches the pattern of a container ID
 func FindContainerID(s string) string {
-	return containerIDPattern.FindString(s)
+	match := containerIDPattern.FindIndex([]byte(s))
```

Comment:
Are we sure this will always be the first match ? Maybe the correct ID is in the second match ?

Reply:
I'm not sure at all nope, I also wonder if it was needed to loop over all the possible matches, even if I don't think it's needed to parse /proc, we definitively could do that for other upcoming use-cases. Would you prefer to add it right now?